components/scream/src/physics/p3/p3_functions_impose_max_total_Ni_impl.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef P3_FUNCTIONS_IMPOSE_MAX_TOTAL_NI_IMPL_HPP
+#define  P3_FUNCTIONS_IMPOSE_MAX_TOTAL_NI_IMPL_HPP
+
+#include "p3_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace p3 {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::impose_max_total_Ni(Spack& nitot_local, const Spack& max_total_Ni, const Spack& inv_rho_local)
+{
+    //--------------------------------------------------------------------------------
+    // Impose maximum total ice number concentration (total of all ice categories).
+    // If the sum of all nitot(:) exceeds maximum allowable, each category to preserve
+    // ratio of number between categories.
+    //--------------------------------------------------------------------------------
+    
+    const auto nitot_not_small = nitot_local >= 1e-20; 
+    if(nitot_not_small.any()){
+        nitot_local = nitot_local*min(max_total_Ni * inv_rho_local/nitot_local, 1.0); 
```

Comment:
Use an int literal here (`1` instead of `1.0`).

Reply:
@jgfouca , why int literal here? Is there a performance reason? Just curious.

Reply:
@pressel , in order to avoid overpromotion. C++ will interpret any literal with a decimal point as a double precision float. This will cause BFB headaches for us in single precision mode (fortran does not work this way). If you really need a float literal, you can use the sp (scream-precision) macro. Using an int means you don't have to worry about it.